mods/edge-hot-corner-desktop-switch.wh.cpp

```diff
@@ -0,0 +1,173 @@
+
+    // Arrow up  (extended + key-up)
+    inputs[3].type = INPUT_KEYBOARD;
+    inputs[3].ki.wVk = vkDir;
+    inputs[3].ki.dwFlags = KEYEVENTF_EXTENDEDKEY | KEYEVENTF_KEYUP;
+
+    // Ctrl up
+    inputs[4].type = INPUT_KEYBOARD;
+    inputs[4].ki.wVk = VK_CONTROL;
+    inputs[4].ki.dwFlags = KEYEVENTF_KEYUP;
+
+    // Win up   (extended + key-up)
+    inputs[5].type = INPUT_KEYBOARD;
+    inputs[5].ki.wVk = VK_LWIN;
+    inputs[5].ki.dwFlags = KEYEVENTF_EXTENDEDKEY | KEYEVENTF_KEYUP;
+
+    // Send them all in one go
+    UINT sent = SendInput(_countof(inputs), inputs, sizeof(INPUT));
+    Wh_Log(L"EdgeHotCorner",
+           L"SimulateDesktopSwitch(%s): SendInput injected %u events",
+           (direction < 0 ? L"Left" : L"Right"), sent);
+}
+
+DWORD WINAPI MonitorThread(LPVOID) {
+    POINT pt;
+    int zone = 0;  // 0 = none, 1 = left, 2 = right
+    DWORD enterTime = 0;
+
+    while (g_Running) {
+        Sleep(20);
+
+        if (!GetCursorPos(&pt))
+            continue;
+
+        int vsX = GetSystemMetrics(SM_XVIRTUALSCREEN);
+        int vsW = GetSystemMetrics(SM_CXVIRTUALSCREEN);
+
+        // Only log if near an edge (within 100 pixels from left or right virtual screen)
+        if (pt.x <= vsX + 100 || pt.x >= (vsX + vsW - 100)) {
+            Wh_Log(L"EdgeHotCorner",
+                   L"Cursor x = %d, virtualScreenStartX = %d, width = %d",
+                   pt.x, vsX, vsW);
+        }
+
+        // Get the monitor under the cursor
+        HMONITOR hMon = MonitorFromPoint(pt, MONITOR_DEFAULTTONULL);
+        if (!hMon) {
+            zone = 0;
+            continue;
+        }
+
+        // Query its full bounds
+        MONITORINFO mi = { sizeof(mi) };
+        GetMonitorInfo(hMon, &mi);
+        int left = mi.rcMonitor.left;
+        int right = mi.rcMonitor.right;
+
+        bool atLeft = (pt.x <= left + settings.edgeWidth);
+        bool atRight = (pt.x >= right - settings.edgeWidth);
+
+        int newZone = atLeft ? 1 : (atRight ? 2 : 0);
+
+        if (newZone != 0) {
+            if (zone != newZone) {
+                // Just entered a hot zone
+                zone = newZone;
+                enterTime = GetTickCount();
+            } else if (GetTickCount() - enterTime >= (DWORD)settings.thresholdMs) {
+                // Held in zone long enough: trigger switch
+                Wh_Log(L"EdgeHotCorner", L"Switching desktop: %s",
+                       (zone == 1 ? L"Previous" : L"Next"));
+                SimulateDesktopSwitch(zone == 1 ? -1 : +1);
+                // Reset so we don’t retrigger immediately
+                zone = 0;
+            }
+        } else {
+            // Left the hot zone; reset
+            zone = 0;
+        }
+    }
+
+    return 0;
+}
+
+BOOL Wh_ModInit() {
+    Wh_Log(L"EdgeHotCorner", L"Initializing mod");
+    LoadSettings();
+
+    // Start the monitoring thread
+    g_Running = true;
+    g_hThread = CreateThread(nullptr, 0, MonitorThread, nullptr, 0, nullptr);
```

Comment:
There can be more than one `explorer.exe` process, e.g. if the "Launch Folder Windows in a Separate Process" option is enabled.
You can make sure the mod is only loaded in the process that owns the taskbar. You can run something like this at the beginning of `Wh_ModInit`:
```cpp
if (!FindCurrentProcessTaskbarWnd()) {
  return FALSE;
}
```
The function code:
https://github.com/m417z/my-windhawk-mods/blob/04f0edd8a7b12d0875c1861a294336d968508d73/mods/icon-resource-redirect.wh.cpp#L2175-L2194